operations-stdlib/src/commonMain/kotlin/string/Replace.kt

```diff
@@ -0,0 +1,86 @@
+package string
+
+import operation.StandardLogicOperation
+import kotlin.runCatching
+import utils.asList
+
+object Replace: StandardLogicOperation, StringUnwrapStrategy {
+    override fun evaluateLogic(expression: Any?, data: Any?): Any? =
+        ReplaceExecutor.buildFromExpression(expression.asList)?.execute()
+}
+
+private data class ReplaceExecutor(
+    val replaceCandidate: String,
+    val oldString: String,
+    val newString: String,
+    val mode: ReplaceMode
+) {
+
+    fun execute() = mode.execute(this)
+
+    companion object {
+        const val REPLACE_CANDIDATE_INDEX = 0
+        const val OLD_STRING_INDEX = 1
+        const val NEW_STRING_INDEX = 2
+        const val COUNT_INDEX = 3
+    }
+}
+private fun ReplaceExecutor.Companion.buildFromExpression(expression: List<Any?>): ReplaceExecutor? {
+    runCatching {
+        val mode = ReplaceMode.build(expression[COUNT_INDEX] as String)
+        ReplaceExecutor(
+            expression[REPLACE_CANDIDATE_INDEX] as String,
+            expression[OLD_STRING_INDEX] as String,
+            expression[NEW_STRING_INDEX] as String,
+            mode
+        )
+
+    }.fold(onSuccess = {
+        return it
+    }, onFailure = {
+        return null
+    })
+}
+
+private sealed class ReplaceMode{
+    abstract fun execute(executor: ReplaceExecutor) : Any?
+
+    companion object {
+        fun build(mode: String): ReplaceMode {
+            when {
+                mode == AllReplace.name -> {
+                    return AllReplace
+                }
+                mode.toIntOrNull() != null -> {
+                    return FewReplace(mode.toInt())
+                }
+            }
+            throw IllegalArgumentException(mode)
+        }
+    }
+}
+
+private object AllReplace : ReplaceMode() {
```

Comment:
i suggest moving the whole selaed class with its inheritors to a separate file, will make it clearer